server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/LogicalTableRecognitionUtils.java

```diff
@@ -0,0 +1,518 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.connection.logicaldatabase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import com.oceanbase.odc.common.lang.Pair;
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.core.shared.exception.UnexpectedException;
+import com.oceanbase.odc.service.connection.logicaldatabase.model.DataNode;
+import com.oceanbase.odc.service.connection.logicaldatabase.model.LogicalTable;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/3/26 14:17
+ * @Description: []
+ */
+public class LogicalTableRecognitionUtils {
```

Comment:
if LogicalTableRecognitionUtils only called from LogicalTableFinder, may use package access privilege

Reply:
Sure, I changed it to the default access privilege.